actors/builtin/miner/miner_test.go

```diff
@@ -1306,7 +1306,9 @@ func TestCCUpgrade(t *testing.T) {
 		oldPower := miner.QAPowerForSector(actor.sectorSize, oldSector)
 		expectedFee := miner.PledgePenaltyForContinuedFault(actor.epochRewardSmooth, actor.epochQAPowerSmooth, oldPower)
 		expectedPowerDelta := miner.NewPowerPairZero()
+		rt.SetNetworkVersion(network.Version5)
```

Comment:
I think in general all unit tests should target the latest network version. If we have a version switch, it will generally be because the prior version is already live in the network, and so we're working to test that the new version will behave as expected. E.g. maybe there's some unexpected interaction between the change guarded by this version and other things that are being tested here.

Which means generally all these tests that are not specifically about the reward vesting should require no diff (because the runtime uses VersionMax). If they do require a diff, that diff will specifically highlight the behaviour that's changing.

Is there a diff here to pledge expectations or something? I think we should update expectations to match (hopefully we can do it just once in the deadlineCron or something).

Reply:
> I think in general all unit tests should target the latest network version.

agree

> Which means generally all these tests that are not specifically about the reward vesting should require no diff

agree on the *should*

> Is there a diff here to pledge expectations or something?

Yes.  The problem is that we use applyRewards as a testing setup function in a bunch of places in a fragile way, not as part of actually testing "what happens if the miners gets a reward message and then x and y happen".  There is a better way to do it as we've known for a while, however designing it and then this refactor seems non-trivial to me.  I didn't want fixing up our testing problems to block this change so I used the ugly approach you see in the diff.  The benefit of this approach is that it makes it really obvious that there is a problem with how we setup tests, because as you point out we shouldn't be using v5 in tests.

Reply:
I have addressed this in #1251 and will rebase this PR upon that one, removing the noise.